### ISSUE TITLE
Error: extendMarkdown() takes 2 positional arguments but 3 were given

### ISSUE THREAD

**Issue by LordOfLuck** [getpelican/pelican-plugins#1246]
I am getting an error when running make html after inluding the render_math plugin 
in my pelicanconf.py. The error:

    ERROR: Could not process posts/random_article.md
      | TypeError: extendMarkdown() takes 2 positional arguments but 3 were given

The pelicanconf contains among other the following lines:

    PLUGIN_PATHS=['pelican-plugins']
    PLUGINS = ['render_math']

pelican-plugins were cloned to root of the project. 
There is only one content file stored in content/posts/ 
I am getting the same error no matter if I use some math or not in the article.
Without the plugin, everything works fine.

I am on mac. This is the output of my pip freeze:

    blinker==1.4
    docutils==0.16
    feedgenerator==1.9
    Jinja2==2.11.1
    Markdown==2.6.8
    MarkupSafe==1.1.1
    pelican==3.6.0
    Pygments==2.5.2
    python-dateutil==2.8.1
    pytz==2019.3
    six==1.14.0
    Unidecode==1.1.1

I also tried pelican 3.7 but had no success either. Any help would be appreciated.

**Comment by arthur-e** [comment]
This has been a huge headache for me as well. I'm not sure when this stopped working as I haven't used Pelican in a few months, but while the current version of `render_math` indeed has only 2 arguments to the `extendMarkdown()` function, [the first version accepted three arguments.](https://github.com/getpelican/pelican-plugins/commit/b2fb21704da97206a4ce12ea2130c99ef348a305#diff-7459f9d7c5df05cd329a48fd20e34483R70) 

What is `md_globals` and why is it no longer accepted? What Pelican code is still passing this third argument?

**Comment by arthur-e** [comment]
[This is the commit where the `md_globals` argument was removed.](https://github.com/getpelican/pelican-plugins/commit/42dc0a6f507335a4bbbf680c6f1b27a69aaff88c#diff-7459f9d7c5df05cd329a48fd20e34483 **Notably, the CI tests failed at this point.**

**Comment by justinmayer** [comment]
Hi @arthur-e. We are in the process of slowly migrating plugins over to the [new organization](https://github.com/pelican-plugins) as individual repositories. The _Render Math_ plugin is one of the few that has already been migrated over. While I have no idea whether trying the migrated version will address this problem, would you try the version from its new individual repository and see whether it is similarly affected? https://github.com/pelican-plugins/render-math

**Comment by arthur-e** [comment]
Hi @justinmayer, thanks for your response. I have placed the new `render-math` into the plugins directory and tried again. Getting a different error, at least!

```
CRITICAL: AttributeError: module 'render-math' has no attribute 'register'
```

I am going to try a fresh plug-ins directory, then try adding the new plug-ins one-by-one, perhaps progressively removing dependencies.

**Comment by justinmayer** [comment]
Sorry for the confusing state of affairs, @arthur-e, and thanks for your patience. `pip install pelican-render-math` will eventually be the proper way to install, but I haven't published the latest version to PyPI yet, so I can't guarantee that will work well for you. Following are the steps I just took to install Pelican master inside a fresh virtual environment, then install the master version of the Render Math plugin, and then successfully build the Render Math test reST file:
```
pip install -e ~/Projects/pelican
pip install -e ~/Projects/pelican-plugins/render-math
mkdir render-math-test-project && cd render-math-test-project
pelican-quickstart
cp ~/Projects/pelican-plugins/render-math/pelican/plugins/render_math/test_data/article_with_math_formulas.rst content/
pelican
```
Does that help point you in the right direction for the moment?

**Comment by justinmayer** [comment]
@arthur-e: I believe `md_globals` was (past tense) part of [Python-Markdown](https://github.com/Python-Markdown/markdown) — not Pelican — and was [deprecated](https://github.com/Python-Markdown/markdown/pull/697) in Python-Markdown v3.0+, which is why @Lucas-C removed it in the commit you referenced. What version of Python-Markdown are you using?

**Comment by arthur-e** [comment]
Is `Python-Markdown` registered simply as `Markdown` in pip? I was using `Markdown==2.6.5` in prior builds of this environment, where `render_math` was failing. **Switching from `Markdown` 2.6.5 to 3.2.1 (any 3.x perhaps?) fixes this issue.** Thanks!

I am currently using the trunk of the [original pelican-plugins repository,](https://github.com/getpelican/pelican-plugins) and with [updated packages as listed above.](https://github.com/getpelican/pelican-plugins/issues/1246#issuecomment-609424753) `pelican-render-math` is *not* installed.

**Comment by justinmayer** [comment]
Yes, the PyPI package name for Python-Markdown is `markdown`, and version 3+ is [indeed required](https://github.com/getpelican/pelican/blob/master/pyproject.toml#L39) for the current Pelican version.

Glad to hear you were able to resolve the problem. If someone else encounters it, please be sure to use the Python-Markdown version specified in Pelican's dependency requirements. 👍

**Comment by justinmayer** [comment]
For anyone interested, I just published Render Math 1.0 to PyPI. 🎉
